default_gateway.go

```diff
@@ -119,15 +130,15 @@ func (sb *sandbox) needDefaultGW() bool {
 
 func (sb *sandbox) getEndpointInGWNetwork() *endpoint {
 	for _, ep := range sb.getConnectedEndpoints() {
-		if ep.getNetwork().name == libnGWNetwork {
+		if ep.getNetwork().name == libnGWNetwork && strings.HasPrefix(ep.Name(), "gateway_") {
```

Comment:
We will take care of this later.
We already have `func (ep *endpoint) endpointInGWNetwork()`, we just need to use it here and have wndows reimplement his version.